src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -629,8 +629,8 @@ private async Task ValidateViewModel(EditTenantViewModel model, bool newTenant)
                 ModelState.AddModelError(nameof(EditTenantViewModel.RequestUrlPrefix), S["Host and url prefix can not be empty at the same time."]);
             }
 
-            var allOtherShells = allSettings.Where(tenant => !string.Equals(tenant.Name, model.Name, StringComparison.OrdinalIgnoreCase));
-            if (allOtherShells.Any(tenant => String.Equals(tenant.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && String.Equals(tenant.RequestUrlHost, model.RequestUrlHost, StringComparison.OrdinalIgnoreCase)))
+            var allOtherShells = allSettings.Where(t => !string.Equals(t.Name, model.Name, StringComparison.OrdinalIgnoreCase));
+            if (allOtherShells.Any(t => String.Equals(t.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && t.RequestUrlHost.Contains(model.RequestUrlHost, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Try the reproduction steps under https://github.com/OrchardCMS/OrchardCore/issues/10269#issue-996369108. You'll get an NRE.

Reply:
I will write a unit test for that instead, it might be `RequestUrlPrefix` or `RequestUrlHost` causes the NRE